packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/events/TouchEventDispatchTest.kt

```diff
@@ -0,0 +1,681 @@
+      createTouchEvent(
+        GESTURE_START_TIME,
+        MotionEvent.ACTION_POINTER_DOWN,
+        1,
+        intArrayOf(0, 1),
+        arrayOf(pointerCoords(1f, 1f), pointerCoords(2f, 1f))
+      ),
+      createTouchEvent(
+        GESTURE_START_TIME,
+        MotionEvent.ACTION_MOVE,
+        0,
+        intArrayOf(0, 1),
+        arrayOf(pointerCoords(1f, 2f), pointerCoords(2f, 1f))
+      ),
+      createTouchEvent(
+        GESTURE_START_TIME,
+        MotionEvent.ACTION_CANCEL,
+        0,
+        intArrayOf(0, 1),
+        arrayOf(pointerCoords(1f, 3f), pointerCoords(2f, 1f))
+      )
+    )
+
+  /** Expected values for [startMoveCancelSequence] */
+  private val startMoveCancelExpectedSequence =
+    listOf(
+      /*
+       * START event for touch 1:
+       * {
+       *   touch: 0,
+       *   touches: [touch1],
+       *   changed: [touch1]
+       * }
+       */
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        1f,
+        1f,
+        GESTURE_START_TIME,
+        0,
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 1f, GESTURE_START_TIME, 0)
+        ),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 1f, GESTURE_START_TIME, 0)
+        )
+      ),
+      /*
+       * START event for touch 2:
+       * {
+       *   touch: 1,
+       *   touches: [touch0, touch1],
+       *   changed: [touch1]
+       * }
+       */
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        2f,
+        1f,
+        GESTURE_START_TIME,
+        1,
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 1f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        ),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        )
+      ),
+      /*
+       * MOVE event for touch 1:
+       * {
+       *   touch: 0,
+       *   touches: [touch0, touch1],
+       *   changed: [touch0, touch1]
+       * }
+       * {
+       *   touch: 1,
+       *   touches: [touch0, touch1],
+       *   changed: [touch0, touch1]
+       * }
+       */
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        1f,
+        2f,
+        GESTURE_START_TIME,
+        0,
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 2f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        ),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 2f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        )
+      ),
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        2f,
+        1f,
+        GESTURE_START_TIME,
+        1,
+        listOf<WritableMap?>(
```

Comment:
```suggestion
        listOf<WritableMap>(
```

This needs to be applied everywhere